system-design/app-interoperability/workload-orchestration-agent.md

```diff
@@ -83,9 +83,214 @@ The following shall be configurable to ensure compliance with local operations.
+|-----------------|-----------------|-----------------|-----------------|
+| pointer           | string          | Y    | Defines the component the previously stated value will be applied to.|
+| components        | array         | Y    | Describes which component in the deployment specification the parameter should be applied to.|
+
+**Deployment specification example for Cluster with multiple components**
+
+```
+apiVersion: margo.deployment/v1
+kind: ApplicationDeployment
+metadata:
+    id: 0f83e682-a35b-4533-ae93-ff3810095f31
+    applicationId: com-northstarida-digitron-orchestrator
+spec:
+    deploymentProfile:
+      type: helm.v3
+      components:
+        - name: digitron-orchestrator
+          properties:
+            repository: oci://northstarida.azurecr.io/charts/northstarida-digitron-orchestrator
+            revision: 1.0.9
+            wait: true
+            timeout: 8m0s
+        - name: database-services
+           properties:
+             repository: oci://quay.io/charts/realtime-database-services
+             revision: 2.3.7
+    parameters:
+      hostname:
+        value: edge.host.local
+        targets:
+          - pointer: edge_host_name
+            components: ["digitron-orchestrator"]
+          - pointer: global.ingress.host
+            components: ["digitron-orchestrator", "database-services"]
+      pollFrequency:
+        value: 30
+        targets: 
+          - pointer: settings.pollFrequency
+            components: ["digitron-orchestrator"]
+      siteId:
+        value: d8b64e67-3bc4-4d0b-a3b8-da94b8f36315
+        targets:
+          - pointer: settings.siteId
+            components: ["digitron-orchestrator"]
+          - pointer: globals.site.id
+            components: ["database-services"]
+      cpuLimit:
+        value: 3
+        targets:
+          - pointer: settings.limits.cpu
+            components: ["digitron-orchestrator"]
+      memoryLimit:
+        value: 16384
+        targets:
+          - pointer: settings.limits.memory
+            components: ["digitron-orchestrator"]
+```
+
+
+**Deployment specification example for Standalone Device with a single component**
+
+> **Investigation Needed:** Further Investigation needed on the content regarding deployment for Docker Compose applications. 
+
+```
+apiVersion: margo.org/v1-alpha1
+kind: ApplicationDeployment
+metadata:
+    id: 0f83e682-a35b-4533-ae93-ff3810095f31
+    applicationId: com-northstarida-digitron-orchestrator
+spec:
+    deploymentProfile:
+  - type: docker-compose
+    components:
+      - name: digitron-orchestrator-docker
+        properties:
+          packageLocation: https://northsitarida.com/digitron/docker/digitron-orchestrator.tar.gz
+          keyLocation: https://northsitarida.com/digitron/docker/public-key.asc
+    parameters:
+      hostname:
+        value: edge.host.local
+        targets:
+          - pointer: edge_host_name
+            components: ["digitron-orchestrator"]
```

Comment:
The component names in the array don't match in the component name in the deploymentProfile.